### PR TITLE
Create and add C4 model documentation

### PR DESCRIPTION
**Pull request by Malaguth**
docs: Add and update the drawio C4 model
- The c4 model is a work in progress (WIP)

docs: Split functional and technical info, also add c4 model page
- Create a folder for functional docs and update the previous links
- Create a folder for technical docs, add a c4 model page, and add links to the readme

### CODE REVIEW

**Review by ViniciusLAAraujo — APPROVED** [review]

File: doc/non_functional/technical_requirements.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+## Technical Debt Management
+
+18. **Code Maintainability:** The system should be designed and implemented in a way that minimizes technical debt, making it easier to maintain and upgrade in the future.
+
+## Training
+
+19. **Ease of Learning:** The system should be easy to learn and use, minimizing the need for extensive training.
+
+##
```
Comment (Malaguth) [inline doc/non_functional/technical_requirements.md:80]: The `##` is used for subtitles on markdown.  Which subtitle did you want to put here? Or should we remove this line?

Comment (Malaguth) [reply doc/non_functional/technical_requirements.md:80]: We have images in dark and light modes. It would be good to define a pattern and follow it.

**Review by Malaguth — COMMENTED** [review]
We have images in dark and light modes. It would be good to define a pattern and follow it.

**Comment by ViniciusLAAraujo** [comment]
> We have images in dark and light modes. It would be good to define a pattern and follow it.

**Comment by ViniciusLAAraujo** [comment]
> We have images in dark and light modes. It would be good to define a pattern and follow it.

I've changed all of them to black background to be the same

**Pull request merged**